CI: Travis failed on arm64: TestService_NotifyWithBlockedSend  panic: runtime error: index out of range [0] with length 0

CI failure Link:
https://travis-ci.com/github/cilium/cilium/jobs/349673356#L7084
https://travis-ci.com/github/cilium/cilium/jobs/350839273#L7075

```
=== RUN   TestService_NotifyWithBlockedSend
panic: runtime error: index out of range [0] with length 0
goroutine 151 [running]:
github.com/cilium/cilium/pkg/hubble/peer.(*buffer).Pop(0x400014ecc0, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/hubble/peer/buffer.go:92 +0x3e4
github.com/cilium/cilium/pkg/hubble/peer.(*Service).Notify.func3(0x0, 0x0)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/hubble/peer/service.go:102 +0x7c
golang.org/x/sync/errgroup.(*Group).Go.func1(0x4000604c00, 0x400038c540)
	/home/travis/gopath/src/github.com/cilium/cilium/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x58
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/travis/gopath/src/github.com/cilium/cilium/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x60
FAIL	github.com/cilium/cilium/pkg/hubble/peer	0.139s
```

--- a/pkg/hubble/peer/buffer.go
+++ b/pkg/hubble/peer/buffer.go
@@ -89,6 +89,13 @@ func (b *buffer) Pop() (*peerpb.ChangeNotification, error) {
 			return nil, io.EOF
 		}
 	}
+	//While waiting for b.mu.Lock, b.buffer may be closed.
+	select {
+	case <-b.stop:
+		b.mu.Unlock()
+		return nil, io.EOF
+	default:
+	}
 	cn := b.buf[0]
 	b.buf[0] = nil
 	b.buf = b.buf[1:]

